CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ endif()
 dgl_option(USE_CUDA "Build with CUDA" OFF)
 dgl_option(USE_OPENMP "Build with OpenMP" ON)
 dgl_option(USE_AVX "Build with AVX optimization" ON)
+dgl_option(USE_AMP "Build with fp16 support to enable AMP(automatic mixed precision)" OFF)
```

Comment:
Disabled by default because we only support CUDA ARCH>=530, while our binary supports sm35 and sm50.
User could manually turn this flag on if they would like to train GNN with AMP.

Reply:
@BarclayII FYI.